yt_dlp/YoutubeDL.py

```diff
@@ -3560,14 +3560,9 @@ def download(self, url_list):
 
         return self._download_retcode
 
-    def download_with_info_file(self, info_filename):
-        with contextlib.closing(fileinput.FileInput(
-                [info_filename], mode='r',
-                openhook=fileinput.hook_encoded('utf-8'))) as f:
-            # FileInput doesn't have a read method, we can't call json.load
-            infos = [self.sanitize_info(info, self.params.get('clean_infojson', True))
-                     for info in variadic(json.loads('\n'.join(f)))]
-        for info in infos:
+    def download_with_info(self, info_list):
```

Comment:
```suggestion
    def download_with_info(self, *info_list):
```
is better API imo

Reply:
Thanks for the tip, I am not well versed in python ways. Implemented.